[ICE]: ` all spans must be disjoint`

<!--
[31mICE[0m: Rustc ./a.rs '' 'assertion `left == right` failed: all spans must be disjoint   '  left: Some([SubstitutionPart { span: /tmp/im/2/a.rs:5:13: 5:16 (#0), snippet: "" }, SubstitutionPart { span: /tmp/im/2/a.rs:5:15: 5:16 (#0), snippet: "/* e */" }])' ' right: None'', 'assertion `left == right` failed: all spans must be disjoint   '  left: Some([SubstitutionPart { span: /tmp/im/2/a.rs:5:13: 5:16 (#0), snippet: "" }, SubstitutionPart { span: /tmp/im/2/a.rs:5:15: 5:16 (#0), snippet: "/* e */" }])' ' right: None''
File: /tmp/im/2/a.rs
-->


snippet:
````rust
struct B;
struct D;
struct F;
fn foo(g: F, y: F, e: &E) {
    foo(B, g, D, E, F, G)
}

fn main() {}

````

Version information
````
rustc 1.95.0-nightly (a18e6d9d1 2026-01-24)
binary: rustc
commit-hash: a18e6d9d1473d9b25581dd04bef6c7577999631c
commit-date: 2026-01-24
host: x86_64-unknown-linux-gnu
release: 1.95.0-nightly
LLVM version: 21.1.8

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs#L332-L344

Command:
`/home/matthias/.rustup/toolchains/alt-master/bin/rustc `

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0425]: cannot find type `E` in this scope
 --> /tmp/icemaker_global_tempdir.zW2Db1LtsPoH/rustc_testrunner_tmpdir_reporting.csYc3qNJrALQ/mvce.rs:4:24
  |
1 | struct B;
  | --------- similarly named struct `B` defined here
...
4 | fn foo(g: F, y: F, e: &E) {
  |                        ^
  |
help: a struct with a similar name exists
  |
4 - fn foo(g: F, y: F, e: &E) {
4 + fn foo(g: F, y: F, e: &B) {
  |
help: you might be missing a type parameter
  |
4 | fn foo<E>(g: F, y: F, e: &E) {
  |       +++

error[E0425]: cannot find value `E` in this scope
 --> /tmp/icemaker_global_tempdir.zW2Db1LtsPoH/rustc_testrunner_tmpdir_reporting.csYc3qNJrALQ/mvce.rs:5:18
  |
5 |     foo(B, g, D, E, F, G)
  |                  ^
  |
help: a local variable with a similar name exists
  |
5 -     foo(B, g, D, E, F, G)
5 +     foo(B, g, D, e, F, G)
  |
help: consider importing one of these constants
  |
1 + use std::f128::consts::E;
  |
1 + use std::f16::consts::E;
  |
1 + use std::f32::consts::E;
  |
1 + use std::f64::consts::E;
  |

error[E0425]: cannot find value `G` in this scope
 --> /tmp/icemaker_global_tempdir.zW2Db1LtsPoH/rustc_testrunner_tmpdir_reporting.csYc3qNJrALQ/mvce.rs:5:24
  |
5 |     foo(B, g, D, E, F, G)
  |                        ^
  |
help: a local variable with a similar name exists
  |
5 -     foo(B, g, D, E, F, G)
5 +     foo(B, g, D, E, F, g)
  |
help: you might be missing a const parameter
  |
4 | fn foo<const G: /* Type */>(g: F, y: F, e: &E) {
  |       +++++++++++++++++++++

error[E0601]: `main` function not found in crate `mvce`
 --> /tmp/icemaker_global_tempdir.zW2Db1LtsPoH/rustc_testrunner_tmpdir_reporting.csYc3qNJrALQ/mvce.rs:6:2
  |
6 | }
  |  ^ consider adding a `main` function to `/tmp/icemaker_global_tempdir.zW2Db1LtsPoH/rustc_testrunner_tmpdir_reporting.csYc3qNJrALQ/mvce.rs`


thread 'rustc' (1103081) panicked at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs:338:29:
assertion `left == right` failed: all spans must be disjoint
  left: Some([SubstitutionPart { span: /tmp/icemaker_global_tempdir.zW2Db1LtsPoH/rustc_testrunner_tmpdir_reporting.csYc3qNJrALQ/mvce.rs:5:13: 5:16 (#0), snippet: "" }, SubstitutionPart { span: /tmp/icemaker_global_tempdir.zW2Db1LtsPoH/rustc_testrunner_tmpdir_reporting.csYc3qNJrALQ/mvce.rs:5:15: 5:16 (#0), snippet: "/* e */" }])
 right: None
stack backtrace:
   0:     0x7efd962c47c0 - std[39b618b54d6f0501]::backtrace_rs::backtrace::libunwind::trace
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/../../backtrace/src/backtrace/libunwind.rs:117:9
   1:     0x7efd962c47c0 - std[39b618b54d6f0501]::backtrace_rs::backtrace::trace_unsynchronized::<std[39b618b54d6f0501]::sys::backtrace::_print_fmt::{closure#1}>
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:14
   2:     0x7efd962c47c0 - std[39b618b54d6f0501]::sys::backtrace::_print_fmt
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/sys/backtrace.rs:74:9
   3:     0x7efd962c47c0 - <<std[39b618b54d6f0501]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[8437160f224edb47]::fmt::Display>::fmt
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/sys/backtrace.rs:44:26
   4:     0x7efd919d40ae - <core[8437160f224edb47]::fmt::rt::Argument>::fmt
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/fmt/rt.rs:152:76
   5:     0x7efd919d40ae - core[8437160f224edb47]::fmt::write
   6:     0x7efd962dcdbc - std[39b618b54d6f0501]::io::default_write_fmt::<std[39b618b54d6f0501]::sys::stdio::unix::Stderr>
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/io/mod.rs:639:11
   7:     0x7efd962dcdbc - <std[39b618b54d6f0501]::sys::stdio::unix::Stderr as std[39b618b54d6f0501]::io::Write>::write_fmt
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/io/mod.rs:1994:13
   8:     0x7efd9628a046 - <std[39b618b54d6f0501]::sys::backtrace::BacktraceLock>::print
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/sys/backtrace.rs:47:9
   9:     0x7efd9628a046 - std[39b618b54d6f0501]::panicking::default_hook::{closure#0}
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/panicking.rs:292:27
  10:     0x7efd962b3981 - std[39b618b54d6f0501]::panicking::default_hook
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/panicking.rs:319:9
  11:     0x7efd92d9d808 - <alloc[7770521358196be1]::boxed::Box<dyn for<'a, 'b> core[8437160f224edb47]::ops::function::Fn<(&'a std[39b618b54d6f0501]::panic::PanicHookInfo<'b>,), Output = ()> + core[8437160f224edb47]::marker::Sync + core[8437160f224edb47]::marker::Send> as core[8437160f224edb47]::ops::function::Fn<(&std[39b618b54d6f0501]::panic::PanicHookInfo,)>>::call
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/alloc/src/boxed.rs:2220:9
  12:     0x7efd92d9d808 - rustc_driver_impl[4a540bf6cb669a4a]::install_ice_hook::{closure#1}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_driver_impl/src/lib.rs:1491:17
  13:     0x7efd92d9d808 - <alloc[7770521358196be1]::boxed::Box<rustc_driver_impl[4a540bf6cb669a4a]::install_ice_hook::{closure#1}> as core[8437160f224edb47]::ops::function::Fn<(&dyn for<'a, 'b> core[8437160f224edb47]::ops::function::Fn<(&'a std[39b618b54d6f0501]::panic::PanicHookInfo<'b>,), Output = ()> + core[8437160f224edb47]::marker::Sync + core[8437160f224edb47]::marker::Send, &std[39b618b54d6f0501]::panic::PanicHookInfo)>>::call
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/alloc/src/boxed.rs:2220:9
  14:     0x7efd92d9d808 - std[39b618b54d6f0501]::panicking::update_hook::<alloc[7770521358196be1]::boxed::Box<rustc_driver_impl[4a540bf6cb669a4a]::install_ice_hook::{closure#1}>>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/panicking.rs:235:47
  15:     0x7efd962b3cfa - <alloc[7770521358196be1]::boxed::Box<dyn for<'a, 'b> core[8437160f224edb47]::ops::function::Fn<(&'a std[39b618b54d6f0501]::panic::PanicHookInfo<'b>,), Output = ()> + core[8437160f224edb47]::marker::Sync + core[8437160f224edb47]::marker::Send> as core[8437160f224edb47]::ops::function::Fn<(&std[39b618b54d6f0501]::panic::PanicHookInfo,)>>::call
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/alloc/src/boxed.rs:2220:9
  16:     0x7efd962b3cfa - std[39b618b54d6f0501]::panicking::panic_with_hook
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/panicking.rs:833:13
  17:     0x7efd9628a10a - std[39b618b54d6f0501]::panicking::panic_handler::{closure#0}
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/panicking.rs:698:13
  18:     0x7efd962756e9 - std[39b618b54d6f0501]::sys::backtrace::__rust_end_short_backtrace::<std[39b618b54d6f0501]::panicking::panic_handler::{closure#0}, !>
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/sys/backtrace.rs:182:18
  19:     0x7efd9628c16d - __rustc[4f67f44b8da499c6]::rust_begin_unwind
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/panicking.rs:689:5
  20:     0x7efd919d4adc - core[8437160f224edb47]::panicking::panic_fmt
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/panicking.rs:80:14
  21:     0x7efd919d49c3 - core[8437160f224edb47]::panicking::assert_failed_inner
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/panicking.rs:434:23
  22:     0x7efd92e3f80c - core[8437160f224edb47]::panicking::assert_failed::<core[8437160f224edb47]::option::Option<&[rustc_errors[16e081fec7c5d5ff]::SubstitutionPart; 2usize]>, core[8437160f224edb47]::option::Option<&[rustc_errors[16e081fec7c5d5ff]::SubstitutionPart; 2usize]>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/panicking.rs:394:5
  23:     0x7efd92e5f744 - <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs:338:29
  24:     0x7efd92e5f744 - core[8437160f224edb47]::iter::adapters::filter_map::filter_map_try_fold::<rustc_errors[16e081fec7c5d5ff]::Substitution, rustc_errors[16e081fec7c5d5ff]::Substitution, alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, core[8437160f224edb47]::result::Result<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, !>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}, alloc[7770521358196be1]::vec::in_place_collect::write_in_place_with_drop<rustc_errors[16e081fec7c5d5ff]::Substitution>::{closure#0}>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/iter/adapters/filter_map.rs:49:28
  25:     0x7efd92e5f744 - core[8437160f224edb47]::iter::adapters::filter::filter_try_fold::<rustc_errors[16e081fec7c5d5ff]::Substitution, alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, core[8437160f224edb47]::result::Result<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, !>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#5}, core[8437160f224edb47]::iter::adapters::filter_map::filter_map_try_fold<rustc_errors[16e081fec7c5d5ff]::Substitution, rustc_errors[16e081fec7c5d5ff]::Substitution, alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, core[8437160f224edb47]::result::Result<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, !>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}, alloc[7770521358196be1]::vec::in_place_collect::write_in_place_with_drop<rustc_errors[16e081fec7c5d5ff]::Substitution>::{closure#0}>::{closure#0}>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/iter/adapters/filter.rs:86:44
  26:     0x7efd92e5f744 - <alloc[7770521358196be1]::vec::into_iter::IntoIter<rustc_errors[16e081fec7c5d5ff]::Substitution> as core[8437160f224edb47]::iter::traits::iterator::Iterator>::try_fold::<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, core[8437160f224edb47]::iter::adapters::filter::filter_try_fold<rustc_errors[16e081fec7c5d5ff]::Substitution, alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, core[8437160f224edb47]::result::Result<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, !>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#5}, core[8437160f224edb47]::iter::adapters::filter_map::filter_map_try_fold<rustc_errors[16e081fec7c5d5ff]::Substitution, rustc_errors[16e081fec7c5d5ff]::Substitution, alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, core[8437160f224edb47]::result::Result<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, !>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}, alloc[7770521358196be1]::vec::in_place_collect::write_in_place_with_drop<rustc_errors[16e081fec7c5d5ff]::Substitution>::{closure#0}>::{closure#0}>::{closure#0}, core[8437160f224edb47]::result::Result<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, !>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/alloc/src/vec/into_iter.rs:357:25
  27:     0x7efd92e5f744 - <core[8437160f224edb47]::iter::adapters::filter::Filter<alloc[7770521358196be1]::vec::into_iter::IntoIter<rustc_errors[16e081fec7c5d5ff]::Substitution>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#5}> as core[8437160f224edb47]::iter::traits::iterator::Iterator>::try_fold::<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, core[8437160f224edb47]::iter::adapters::filter_map::filter_map_try_fold<rustc_errors[16e081fec7c5d5ff]::Substitution, rustc_errors[16e081fec7c5d5ff]::Substitution, alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, core[8437160f224edb47]::result::Result<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, !>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}, alloc[7770521358196be1]::vec::in_place_collect::write_in_place_with_drop<rustc_errors[16e081fec7c5d5ff]::Substitution>::{closure#0}>::{closure#0}, core[8437160f224edb47]::result::Result<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, !>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/iter/adapters/filter.rs:157:19
  28:     0x7efd92e5f744 - <core[8437160f224edb47]::iter::adapters::filter_map::FilterMap<core[8437160f224edb47]::iter::adapters::filter::Filter<alloc[7770521358196be1]::vec::into_iter::IntoIter<rustc_errors[16e081fec7c5d5ff]::Substitution>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#5}>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}> as core[8437160f224edb47]::iter::traits::iterator::Iterator>::try_fold::<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, alloc[7770521358196be1]::vec::in_place_collect::write_in_place_with_drop<rustc_errors[16e081fec7c5d5ff]::Substitution>::{closure#0}, core[8437160f224edb47]::result::Result<alloc[7770521358196be1]::vec::in_place_drop::InPlaceDrop<rustc_errors[16e081fec7c5d5ff]::Substitution>, !>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/iter/adapters/filter_map.rs:138:19
  29:     0x7efd92e5f744 - <core[8437160f224edb47]::iter::adapters::filter_map::FilterMap<core[8437160f224edb47]::iter::adapters::filter::Filter<alloc[7770521358196be1]::vec::into_iter::IntoIter<rustc_errors[16e081fec7c5d5ff]::Substitution>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#5}>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}> as alloc[7770521358196be1]::vec::in_place_collect::SpecInPlaceCollect<rustc_errors[16e081fec7c5d5ff]::Substitution, core[8437160f224edb47]::iter::adapters::filter_map::FilterMap<core[8437160f224edb47]::iter::adapters::filter::Filter<alloc[7770521358196be1]::vec::into_iter::IntoIter<rustc_errors[16e081fec7c5d5ff]::Substitution>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#5}>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}>>>::collect_in_place
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/alloc/src/vec/in_place_collect.rs:378:18
  30:     0x7efd92e5f744 - alloc[7770521358196be1]::vec::in_place_collect::from_iter_in_place::<core[8437160f224edb47]::iter::adapters::filter_map::FilterMap<core[8437160f224edb47]::iter::adapters::filter::Filter<alloc[7770521358196be1]::vec::into_iter::IntoIter<rustc_errors[16e081fec7c5d5ff]::Substitution>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#5}>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}>, rustc_errors[16e081fec7c5d5ff]::Substitution>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/alloc/src/vec/in_place_collect.rs:269:9
  31:     0x7efd92e5f744 - <alloc[7770521358196be1]::vec::Vec<rustc_errors[16e081fec7c5d5ff]::Substitution> as alloc[7770521358196be1]::vec::spec_from_iter::SpecFromIter<rustc_errors[16e081fec7c5d5ff]::Substitution, core[8437160f224edb47]::iter::adapters::filter_map::FilterMap<core[8437160f224edb47]::iter::adapters::filter::Filter<alloc[7770521358196be1]::vec::into_iter::IntoIter<rustc_errors[16e081fec7c5d5ff]::Substitution>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#5}>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}>>>::from_iter
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/alloc/src/vec/in_place_collect.rs:245:9
  32:     0x7efd92e5f744 - <alloc[7770521358196be1]::vec::Vec<rustc_errors[16e081fec7c5d5ff]::Substitution> as core[8437160f224edb47]::iter::traits::collect::FromIterator<rustc_errors[16e081fec7c5d5ff]::Substitution>>::from_iter::<core[8437160f224edb47]::iter::adapters::filter_map::FilterMap<core[8437160f224edb47]::iter::adapters::filter::Filter<alloc[7770521358196be1]::vec::into_iter::IntoIter<rustc_errors[16e081fec7c5d5ff]::Substitution>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#5}>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/alloc/src/vec/mod.rs:3801:9
  33:     0x7efd92e5f744 - <core[8437160f224edb47]::iter::adapters::filter_map::FilterMap<core[8437160f224edb47]::iter::adapters::filter::Filter<alloc[7770521358196be1]::vec::into_iter::IntoIter<rustc_errors[16e081fec7c5d5ff]::Substitution>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#5}>, <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default::{closure#6}> as core[8437160f224edb47]::iter::traits::iterator::Iterator>::collect::<alloc[7770521358196be1]::vec::Vec<rustc_errors[16e081fec7c5d5ff]::Substitution>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/iter/traits/iterator.rs:2035:9
  34:     0x7efd92e5f744 - <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter>::emit_messages_default
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs:366:26
  35:     0x7efd92e8c340 - <rustc_errors[16e081fec7c5d5ff]::annotate_snippet_emitter_writer::AnnotateSnippetEmitter as rustc_errors[16e081fec7c5d5ff]::emitter::Emitter>::emit_diagnostic
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs:93:14
  36:     0x7efd92e5342a - <rustc_errors[16e081fec7c5d5ff]::DiagCtxtInner>::emit_diagnostic::{closure#3}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_errors/src/lib.rs:1365:30
  37:     0x7efd93bf2e14 - rustc_interface[fa053fcf6d9c2fa2]::callbacks::track_diagnostic::<core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>::{closure#0}::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_interface/src/callbacks.rs:49:46
  38:     0x7efd93bf2e14 - rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::enter_context::<rustc_interface[fa053fcf6d9c2fa2]::callbacks::track_diagnostic<core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/ty/context/tls.rs:60:9
  39:     0x7efd93bf2e14 - <std[39b618b54d6f0501]::thread::local::LocalKey<core[8437160f224edb47]::cell::Cell<*const ()>>>::try_with::<rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::enter_context<rustc_interface[fa053fcf6d9c2fa2]::callbacks::track_diagnostic<core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>::{closure#0}, core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/thread/local.rs:513:12
  40:     0x7efd93bf2e14 - <std[39b618b54d6f0501]::thread::local::LocalKey<core[8437160f224edb47]::cell::Cell<*const ()>>>::with::<rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::enter_context<rustc_interface[fa053fcf6d9c2fa2]::callbacks::track_diagnostic<core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>::{closure#0}, core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/thread/local.rs:477:20
  41:     0x7efd93bf2e14 - rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::enter_context::<rustc_interface[fa053fcf6d9c2fa2]::callbacks::track_diagnostic<core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/ty/context/tls.rs:57:9
  42:     0x7efd93bf2e14 - rustc_interface[fa053fcf6d9c2fa2]::callbacks::track_diagnostic::<core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_interface/src/callbacks.rs:49:13
  43:     0x7efd93bf2e14 - rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::with_context_opt::<rustc_interface[fa053fcf6d9c2fa2]::callbacks::track_diagnostic<core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>::{closure#0}, core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/ty/context/tls.rs:79:18
  44:     0x7efd93bf2e14 - rustc_interface[fa053fcf6d9c2fa2]::callbacks::track_diagnostic::<core[8437160f224edb47]::option::Option<rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_interface/src/callbacks.rs:43:5
  45:     0x7efd92e6d0b5 - <rustc_errors[16e081fec7c5d5ff]::DiagCtxtInner>::emit_diagnostic
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_errors/src/lib.rs:1320:9
  46:     0x7efd92e678a9 - <rustc_errors[16e081fec7c5d5ff]::DiagCtxtHandle>::emit_diagnostic
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_errors/src/lib.rs:869:33
  47:     0x7efd92e7c031 - <rustc_errors[16e081fec7c5d5ff]::diagnostic::Diag>::emit_producing_error_guaranteed
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_errors/src/diagnostic.rs:1352:29
  48:     0x7efd92e7c031 - <rustc_span[cb1c21e189ad7a9f]::ErrorGuaranteed as rustc_errors[16e081fec7c5d5ff]::diagnostic::EmissionGuarantee>::emit_producing_guarantee
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_errors/src/diagnostic.rs:42:14
  49:     0x7efd937ce339 - <rustc_errors[16e081fec7c5d5ff]::diagnostic::Diag>::emit
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_errors/src/diagnostic.rs:1359:9
  50:     0x7efd937ce339 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::report_arg_errors
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:712:13
  51:     0x7efd937ce339 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_argument_types
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:562:18
  52:     0x7efd9372a628 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::confirm_builtin_call
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/callee.rs:570:14
  53:     0x7efd93701136 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_call
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/callee.rs:131:22
  54:     0x7efd93701136 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/expr.rs:388:50
  55:     0x7efd937257e7 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/expr.rs:291:23
  56:     0x7efd937257e7 - stacker[3e3eb2a5da8370fd]::maybe_grow::<rustc_middle[d88d6cd5cfb0dbbd]::ty::Ty, <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /rust/deps/stacker-0.1.21/src/lib.rs:57:9
  57:     0x7efd937257e7 - rustc_data_structures[8c00556bcda1036a]::stack::ensure_sufficient_stack::<rustc_middle[d88d6cd5cfb0dbbd]::ty::Ty, <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_data_structures/src/stack.rs:21:5
  58:     0x7efd937257e7 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/expr.rs:286:18
  59:     0x7efd9372309c - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/expr.rs:231:14
  60:     0x7efd937bfd19 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_block::{closure#0}::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1049:49
  61:     0x7efd937bfd19 - <core[8437160f224edb47]::option::Option<&rustc_hir[b25c81c33cbee7a8]::hir::Expr>>::map::<(&rustc_hir[b25c81c33cbee7a8]::hir::Expr, rustc_middle[d88d6cd5cfb0dbbd]::ty::Ty), <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_block::{closure#0}::{closure#0}>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/option.rs:1165:29
  62:     0x7efd937bfd19 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_block::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1049:26
  63:     0x7efd937bfd19 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::with_breakable_ctxt::<<rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_block::{closure#0}, ()>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs:1492:22
  64:     0x7efd937bfd19 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_block
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1041:31
  65:     0x7efd936f4d06 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/expr.rs:387:46
  66:     0x7efd937257e7 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/expr.rs:291:23
  67:     0x7efd937257e7 - stacker[3e3eb2a5da8370fd]::maybe_grow::<rustc_middle[d88d6cd5cfb0dbbd]::ty::Ty, <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /rust/deps/stacker-0.1.21/src/lib.rs:57:9
  68:     0x7efd937257e7 - rustc_data_structures[8c00556bcda1036a]::stack::ensure_sufficient_stack::<rustc_middle[d88d6cd5cfb0dbbd]::ty::Ty, <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_data_structures/src/stack.rs:21:5
  69:     0x7efd937257e7 - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/expr.rs:286:18
  70:     0x7efd9372309c - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/expr.rs:231:14
  71:     0x7efd937229ba - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_expr_with_hint
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/expr.rs:186:14
  72:     0x7efd937229ba - <rustc_hir_typeck[98395963c5ff1a67]::fn_ctxt::FnCtxt>::check_return_or_body_tail
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/expr.rs:968:35
  73:     0x7efd9389ff57 - rustc_hir_typeck[98395963c5ff1a67]::check::check_fn
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/check.rs:130:9
  74:     0x7efd936c4c5d - rustc_hir_typeck[98395963c5ff1a67]::typeck_with_inspect::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/lib.rs:180:9
  75:     0x7efd936e2b1b - rustc_hir_typeck[98395963c5ff1a67]::typeck_with_inspect
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_typeck/src/lib.rs:105:1
  76:     0x7efd95019d1e - rustc_query_impl[21f19dbf574acbb3]::query_impl::typeck::dynamic_query::{closure#2}::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_impl/src/plumbing.rs:282:9
  77:     0x7efd95019d1e - rustc_query_impl[21f19dbf574acbb3]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[21f19dbf574acbb3]::query_impl::typeck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_impl/src/plumbing.rs:552:18
  78:     0x7efd9547986b - rustc_query_impl[21f19dbf574acbb3]::query_impl::typeck::dynamic_query::{closure#2}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_impl/src/plumbing.rs:629:25
  79:     0x7efd9547986b - <rustc_query_impl[21f19dbf574acbb3]::query_impl::typeck::dynamic_query::{closure#2} as core[8437160f224edb47]::ops::function::FnOnce<(rustc_middle[d88d6cd5cfb0dbbd]::ty::context::TyCtxt, rustc_span[cb1c21e189ad7a9f]::def_id::LocalDefId)>>::call_once
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/ops/function.rs:250:5
  80:     0x7efd951176ea - <rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false> as rustc_query_system[b0415c2dd96ffa58]::query::config::QueryConfig<rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>>::compute
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_impl/src/lib.rs:116:9
  81:     0x7efd951176ea - rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr::<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_system/src/query/plumbing.rs:497:72
  82:     0x7efd951176ea - rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::enter_context::<rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/ty/context/tls.rs:60:9
  83:     0x7efd951176ea - <std[39b618b54d6f0501]::thread::local::LocalKey<core[8437160f224edb47]::cell::Cell<*const ()>>>::try_with::<rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::enter_context<rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/thread/local.rs:513:12
  84:     0x7efd951176ea - <std[39b618b54d6f0501]::thread::local::LocalKey<core[8437160f224edb47]::cell::Cell<*const ()>>>::with::<rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::enter_context<rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/thread/local.rs:477:20
  85:     0x7efd951176ea - rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::enter_context::<rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/ty/context/tls.rs:57:9
  86:     0x7efd951176ea - <rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt as rustc_query_system[b0415c2dd96ffa58]::query::QueryContext>::start_query::<rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_impl/src/plumbing.rs:156:13
  87:     0x7efd951176ea - rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::with_related_context::<<rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt as rustc_query_system[b0415c2dd96ffa58]::query::QueryContext>::start_query<rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/ty/context/tls.rs:112:9
  88:     0x7efd951176ea - rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::with_context::<rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::with_related_context<<rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt as rustc_query_system[b0415c2dd96ffa58]::query::QueryContext>::start_query<rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/ty/context/tls.rs:90:36
  89:     0x7efd951176ea - rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::with_context_opt::<rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::with_context<rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::with_related_context<<rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt as rustc_query_system[b0415c2dd96ffa58]::query::QueryContext>::start_query<rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/ty/context/tls.rs:79:18
  90:     0x7efd951176ea - rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::with_context::<rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::with_related_context<<rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt as rustc_query_system[b0415c2dd96ffa58]::query::QueryContext>::start_query<rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/ty/context/tls.rs:90:5
  91:     0x7efd951176ea - rustc_middle[d88d6cd5cfb0dbbd]::ty::context::tls::with_related_context::<<rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt as rustc_query_system[b0415c2dd96ffa58]::query::QueryContext>::start_query<rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/ty/context/tls.rs:103:5
  92:     0x7efd951176ea - <rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt as rustc_query_system[b0415c2dd96ffa58]::query::QueryContext>::start_query::<rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_impl/src/plumbing.rs:142:9
  93:     0x7efd951176ea - rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job_non_incr::<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_system/src/query/plumbing.rs:497:22
  94:     0x7efd951176ea - rustc_query_system[b0415c2dd96ffa58]::query::plumbing::execute_job::<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt, false>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_system/src/query/plumbing.rs:433:9
  95:     0x7efd951176ea - rustc_query_system[b0415c2dd96ffa58]::query::plumbing::try_execute_query::<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_hir_id[fc38eb91a8ee5042]::OwnerId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt, false>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_system/src/query/plumbing.rs:376:13
  96:     0x7efd9528efcc - rustc_query_system[b0415c2dd96ffa58]::query::plumbing::get_query_non_incr::<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_span[cb1c21e189ad7a9f]::def_id::LocalDefId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_system/src/query/plumbing.rs:818:32
  97:     0x7efd9528efcc - stacker[3e3eb2a5da8370fd]::maybe_grow::<rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::query::plumbing::get_query_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_span[cb1c21e189ad7a9f]::def_id::LocalDefId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}>
                               at /rust/deps/stacker-0.1.21/src/lib.rs:57:9
  98:     0x7efd9528efcc - rustc_data_structures[8c00556bcda1036a]::stack::ensure_sufficient_stack::<rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::query::plumbing::get_query_non_incr<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_span[cb1c21e189ad7a9f]::def_id::LocalDefId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>::{closure#0}>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_data_structures/src/stack.rs:21:5
  99:     0x7efd9528efcc - rustc_query_system[b0415c2dd96ffa58]::query::plumbing::get_query_non_incr::<rustc_query_impl[21f19dbf574acbb3]::DynamicConfig<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_span[cb1c21e189ad7a9f]::def_id::LocalDefId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[21f19dbf574acbb3]::plumbing::QueryCtxt>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_system/src/query/plumbing.rs:818:5
 100:     0x7efd9528efcc - rustc_query_impl[21f19dbf574acbb3]::query_impl::typeck::get_query_non_incr::__rust_end_short_backtrace
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_query_impl/src/plumbing.rs:605:26
 101:     0x7efd931cfe3e - rustc_middle[d88d6cd5cfb0dbbd]::query::inner::query_ensure::<rustc_data_structures[8c00556bcda1036a]::vec_cache::VecCache<rustc_span[cb1c21e189ad7a9f]::def_id::LocalDefId, rustc_middle[d88d6cd5cfb0dbbd]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[b0415c2dd96ffa58]::dep_graph::graph::DepNodeIndex>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/query/inner.rs:51:9
 102:     0x7efd931cfe3e - <rustc_middle[d88d6cd5cfb0dbbd]::query::plumbing::TyCtxtEnsureOk>::typeck::<rustc_span[cb1c21e189ad7a9f]::def_id::LocalDefId>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/query/plumbing.rs:173:9
 103:     0x7efd931cfe3e - rustc_hir_analysis[bc0a96f402c1d791]::check_crate::{closure#2}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_hir_analysis/src/lib.rs:254:29
 104:     0x7efd931cfe3e - <rustc_middle[d88d6cd5cfb0dbbd]::ty::context::TyCtxt>::par_hir_body_owners::<rustc_hir_analysis[bc0a96f402c1d791]::check_crate::{closure#2}>::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_middle/src/hir/map.rs:339:79
 105:     0x7efd93213e3d - rustc_data_structures[8c00556bcda1036a]::sync::parallel::par_for_each_in::<&rustc_span[cb1c21e189ad7a9f]::def_id::LocalDefId, &[rustc_span[cb1c21e189ad7a9f]::def_id::LocalDefId], <rustc_middle[d88d6cd5cfb0dbbd]::ty::context::TyCtxt>::par_hir_body_owners<rustc_hir_analysis[bc0a96f402c1d791]::check_crate::{closure#2}>::{closure#0}>::{closure#0}::{closure#1}::{closure#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/compiler/rustc_data_structures/src/sync/parallel.rs:181:30
 106:     0x7efd93213e3d - <core[8437160f224edb47]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[8c00556bcda1036a]::sync::parallel::par_for_each_in<&rustc_span[cb1c21e189ad7a9f]::def_id::LocalDefId, &[rustc_span[cb1c21e189ad7a9f]::def_id::LocalDefId], <rustc_middle[d88d6cd5cfb0dbbd]::ty::context::TyCtxt>::par_hir_body_owners<rustc_hir_analysis[bc0a96f402c1d791]::check_crate::{closure#2}>::{closure#0}>::{closure#0}::{closure#1}::{closure#0}> as core[8437160f224edb47]::ops::function::FnOnce<()>>::call_once
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/panic/unwind_safe.rs:274:9


< snip > 

 174:     0x7efd92daa253 - <core[8437160f224edb47]::panic::unwind_safe::AssertUnwindSafe<std[39b618b54d6f0501]::thread::lifecycle::spawn_unchecked<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_with_globals<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_pool_with_globals<rustc_interface[fa053fcf6d9c2fa2]::interface::run_compiler<(), rustc_driver_impl[4a540bf6cb669a4a]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}> as core[8437160f224edb47]::ops::function::FnOnce<()>>::call_once
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/panic/unwind_safe.rs:274:9
 175:     0x7efd92daa253 - std[39b618b54d6f0501]::panicking::catch_unwind::do_call::<core[8437160f224edb47]::panic::unwind_safe::AssertUnwindSafe<std[39b618b54d6f0501]::thread::lifecycle::spawn_unchecked<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_with_globals<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_pool_with_globals<rustc_interface[fa053fcf6d9c2fa2]::interface::run_compiler<(), rustc_driver_impl[4a540bf6cb669a4a]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/panicking.rs:581:40
 176:     0x7efd92daa253 - std[39b618b54d6f0501]::panicking::catch_unwind::<(), core[8437160f224edb47]::panic::unwind_safe::AssertUnwindSafe<std[39b618b54d6f0501]::thread::lifecycle::spawn_unchecked<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_with_globals<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_pool_with_globals<rustc_interface[fa053fcf6d9c2fa2]::interface::run_compiler<(), rustc_driver_impl[4a540bf6cb669a4a]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/panicking.rs:544:19
 177:     0x7efd92daa253 - std[39b618b54d6f0501]::panic::catch_unwind::<core[8437160f224edb47]::panic::unwind_safe::AssertUnwindSafe<std[39b618b54d6f0501]::thread::lifecycle::spawn_unchecked<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_with_globals<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_pool_with_globals<rustc_interface[fa053fcf6d9c2fa2]::interface::run_compiler<(), rustc_driver_impl[4a540bf6cb669a4a]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/panic.rs:359:14
 178:     0x7efd92daa253 - std[39b618b54d6f0501]::thread::lifecycle::spawn_unchecked::<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_with_globals<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_pool_with_globals<rustc_interface[fa053fcf6d9c2fa2]::interface::run_compiler<(), rustc_driver_impl[4a540bf6cb669a4a]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/thread/lifecycle.rs:89:26
 179:     0x7efd92daa253 - <std[39b618b54d6f0501]::thread::lifecycle::spawn_unchecked<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_with_globals<rustc_interface[fa053fcf6d9c2fa2]::util::run_in_thread_pool_with_globals<rustc_interface[fa053fcf6d9c2fa2]::interface::run_compiler<(), rustc_driver_impl[4a540bf6cb669a4a]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[8437160f224edb47]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
                               at /rustc-dev/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/core/src/ops/function.rs:250:5
 180:     0x7efd962c15ef - <alloc[7770521358196be1]::boxed::Box<dyn core[8437160f224edb47]::ops::function::FnOnce<(), Output = ()> + core[8437160f224edb47]::marker::Send> as core[8437160f224edb47]::ops::function::FnOnce<()>>::call_once
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/alloc/src/boxed.rs:2206:9
 181:     0x7efd962c15ef - <std[39b618b54d6f0501]::sys::thread::unix::Thread>::new::thread_start
                               at /rustc/a18e6d9d1473d9b25581dd04bef6c7577999631c/library/std/src/sys/thread/unix.rs:127:17
 182:     0x7efd8fe9698b - <unknown>
 183:     0x7efd8ff1a9cc - <unknown>
 184:                0x0 - <unknown>

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.95.0-nightly (a18e6d9d1 2026-01-24) running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [typeck] type-checking `foo`
#1 [analysis] running analysis passes on crate `mvce`
end of query stack
error: aborting due to 5 previous errors

Some errors have detailed explanations: E0061, E0425, E0601.
For more information about an error, try `rustc --explain E0061`.

```

</p>
</details>

<!--
query stack:
#0 [typeck] type-checking `foo`
#1 [analysis] running analysis passes on crate `mvce`

-->

--- a/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs
+++ b/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs
@@ -2647,7 +2647,15 @@ impl<'a, 'b, 'tcx> FnCallDiagCtxt<'a, 'b, 'tcx> {
                 // To suggest a multipart suggestion when encountering `foo(1, "")` where the def
                 // was `fn foo(())`.
                 let (_, expected_ty) = self.formal_and_expected_inputs[expected_idx];
-                suggestions.push((*arg_span, self.ty_to_snippet(expected_ty, expected_idx)));
+                // Check if the new suggestion would overlap with any existing suggestion.
+                // This can happen when we have both removal suggestions (which may include
+                // adjacent commas) and type replacement suggestions for the same span.
+                let dominated = suggestions
+                    .iter()
+                    .any(|(span, _)| span.contains(*arg_span) || arg_span.overlaps(*span));
+                if !dominated {
+                    suggestions.push((*arg_span, self.ty_to_snippet(expected_ty, expected_idx)));
+                }
             }
         }
     }

--- a/tests/ui/argument-suggestions/disjoint-spans-issue-151607.rs
+++ b/tests/ui/argument-suggestions/disjoint-spans-issue-151607.rs
@@ -1,0 +1,16 @@
+// Regression test for #151607
+// The ICE was "all spans must be disjoint" when emitting diagnostics
+// with overlapping suggestion spans.
+
+struct B;
+struct D;
+struct F;
+fn foo(g: F, y: F, e: &E) {
+    //~^ ERROR cannot find type `E` in this scope
+    foo(B, g, D, E, F, G)
+    //~^ ERROR this function takes 3 arguments but 6 arguments were supplied
+    //~| ERROR cannot find value `E` in this scope
+    //~| ERROR cannot find value `G` in this scope
+}
+
+fn main() {}

--- a/tests/ui/argument-suggestions/disjoint-spans-issue-151607.stderr
+++ b/tests/ui/argument-suggestions/disjoint-spans-issue-151607.stderr
@@ -1,0 +1,85 @@
+error[E0425]: cannot find type `E` in this scope
+  --> $DIR/disjoint-spans-issue-151607.rs:8:24
+   |
+LL | struct B;
+   | --------- similarly named struct `B` defined here
+...
+LL | fn foo(g: F, y: F, e: &E) {
+   |                        ^
+   |
+help: a struct with a similar name exists
+   |
+LL - fn foo(g: F, y: F, e: &E) {
+LL + fn foo(g: F, y: F, e: &B) {
+   |
+help: you might be missing a type parameter
+   |
+LL | fn foo<E>(g: F, y: F, e: &E) {
+   |       +++
+
+error[E0425]: cannot find value `E` in this scope
+  --> $DIR/disjoint-spans-issue-151607.rs:10:18
+   |
+LL |     foo(B, g, D, E, F, G)
+   |                  ^
+   |
+help: a local variable with a similar name exists
+   |
+LL -     foo(B, g, D, E, F, G)
+LL +     foo(B, g, D, e, F, G)
+   |
+help: consider importing one of these constants
+   |
+LL + use std::f128::consts::E;
+   |
+LL + use std::f16::consts::E;
+   |
+LL + use std::f32::consts::E;
+   |
+LL + use std::f64::consts::E;
+   |
+
+error[E0425]: cannot find value `G` in this scope
+  --> $DIR/disjoint-spans-issue-151607.rs:10:24
+   |
+LL |     foo(B, g, D, E, F, G)
+   |                        ^
+   |
+help: a local variable with a similar name exists
+   |
+LL -     foo(B, g, D, E, F, G)
+LL +     foo(B, g, D, E, F, g)
+   |
+help: you might be missing a const parameter
+   |
+LL | fn foo<const G: /* Type */>(g: F, y: F, e: &E) {
+   |       +++++++++++++++++++++
+
+error[E0061]: this function takes 3 arguments but 6 arguments were supplied
+  --> $DIR/disjoint-spans-issue-151607.rs:10:5
+   |
+LL |     foo(B, g, D, E, F, G)
+   |     ^^^ -     -  - unexpected argument #4
+   |         |     |
+   |         |     unexpected argument #3 of type `D`
+   |         unexpected argument #1 of type `B`
+   |
+note: function defined here
+  --> $DIR/disjoint-spans-issue-151607.rs:8:4
+   |
+LL | fn foo(g: F, y: F, e: &E) {
+   |    ^^^             -----
+help: consider borrowing here
+   |
+LL |     foo(B, g, D, E, F, &G)
+   |                        +
+help: remove the extra arguments
+   |
+LL -     foo(B, g, D, E, F, G)
+LL +     foo(, g, F, G)
+   |
+
+error: aborting due to 4 previous errors
+
+Some errors have detailed explanations: E0061, E0425.
+For more information about an error, try `rustc --explain E0061`.

